testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -10105,9 +10105,23 @@ fn skip_mining_long_tx() {
     assert_eq!(sender_2_nonce, 0);
     assert_eq!(sender_1_nonce, 4);
 
-    // Check that we aren't missing burn blocks
+    // Check that we aren't missing burn blocks (except during the Nakamoto transition)
+    let epoch_3 = &naka_conf.burnchain.epochs.unwrap()[StacksEpochId::Epoch30];
     let bhh = u64::from(tip.burn_header_height);
-    test_observer::contains_burn_block_range(220..=bhh).unwrap();
+    let missing = test_observer::get_missing_burn_blocks(220..=bhh).unwrap();
+    let missing_is_error: Vec<_> = missing
+        .into_iter()
+        .filter(|&i| {
+            (i != epoch_3.start_height - 1) || {
+                warn!("Missing burn block {} at epoch 3 transition", i);
+                false
+            }
+        })
+        .collect();
+
+    if !missing_is_error.is_empty() {
+        panic!("Missing the following burn blocks: {missing_is_error:?}");
+    }
```

Comment:
Can you move these checks into a `fn check_nakamoto_no_missing_blocks(conf: &Config)` function? That way, its just reusable between at least these two tests.